custom_components/sat/sensor.py

```
@@ -42,20 +41,105 @@ async def async_setup_entry(_hass: HomeAssistant, _config_entry: ConfigEntry, _a
     _async_add_entities([
-        SatFlameSensor(coordinator, _config_entry),
+        SatCycleSensor(coordinator, _config_entry),
         SatBoilerSensor(coordinator, _config_entry),
         SatManufacturerSensor(coordinator, _config_entry),
+        SatPidSensor(coordinator, _config_entry, climate),
         SatErrorValueSensor(coordinator, _config_entry, climate),
+        SatRequestedSetpoint(coordinator, _config_entry, climate),
         SatHeatingCurveSensor(coordinator, _config_entry, climate),
     ])
 
+    for entity_id in _config_entry.data.get(CONF_ROOMS) or []:
```

Ruff (0.14.8)

`CONF_ROOMS` may be undefined, or defined from star imports

(F405)